collagraph/__init__.py

```diff
@@ -170,6 +183,31 @@ def perform_unit_of_work(self, fiber: Fiber) -> Optional[Fiber]:
                 return sibling
             next_fiber = next_fiber.parent
 
+    def update_class_component(self, fiber: Fiber):
+        component = fiber.component or fiber.alternate and fiber.alternate.component
+        if not component:
+            component = fiber.type(fiber.props)
+
+        # Attach the component instance to the fiber
+        fiber.component = component
+        fiber.component_watcher = watch(
+            # FIXME: sometimes fiber.component is None
+            # Seems to happen after a while after generating lots of changes.
+            # Could be that it is only when updates are coming in faster than
+            # they are rendered...
+            lambda: fiber.component and fiber.component.state,
```

Comment:
Might also need to watch `fiber.component.props`?

Reply:
Just the state should be sufficient.

Reply:
If the props change, that is always caused somewhere higher up the component tree by a state change.

Reply:
Yes, exactly. Should be sufficient like this!